packages/react-core/src/components/Page/Page.tsx

```diff
@@ -28,6 +28,8 @@ export interface PageProps extends React.HTMLProps<HTMLDivElement> {
   role?: string;
   /** an id to use for the [role="main"] element */
   mainContainerId?: string;
+  /** tabIndex to use for the [role="main"] element */
+  mainTabIndex?: number;
```

Comment:
Should this be `number | null`? To unset the attribute from the DOM users have to supply `null` right?

Reply:
`undefined` works as well I believe. I can verify this in a few hours.

Reply:
`undefined` (or just not passing the prop) also works to unset the attribute from the DOM. `null` can be passed as well if strict null checks are disabled by the consumer.

Reply:
I don't see this working as you describe in https://patternfly-react-pr-4181.surge.sh/documentation/react/components/page or locally within react-integration.

Unless I provide ` mainTabIndex={null}` - the tabindex attr is set on <main> with a value of -1, any idea what I'm missing?

Reply:
TLDR; I updated the PR with your suggestion.

I'm the one missing something, not you! This is due to how default properties are set and I've demoed it in https://codesandbox.io/s/upbeat-bogdan-0fyvl . Basically, you can't set a prop explicitly to `undefined` because the way default props work is setting the property if it `=== undefined` rather than `!Object.hasOwnProperty(this, propertyName)`. This is just another case of Javascript sucking:
![image](https://user-images.githubusercontent.com/47335686/81102776-1a19be00-8ede-11ea-8da9-a34d39fc663d.png)
I would expect the call `test({a: undefined})` to print `undefined`, but in fact I don't think it's possible at all.